### ISSUE TITLE
"The dependency Microsoft.Bcl >= 1.1.9 could not be resolved" on "dotnet build"

### ISSUE THREAD

**Issue by cwe1ss** [dotnet/cli#2835]
`dotnet build` (1.0.0-rc2-002610) for the following project.json fails with error `error NU1001: The dependency Microsoft.Bcl >= 1.1.9 could not be resolved.`

```json
{
  "version": "1.0.0-*",
  "buildOptions": {
    "warningsAsErrors": true
  },
  "dependencies": {
    "Microsoft.Bcl": "1.1.9"
  },
  "frameworks": {
    "net46": { }
  }
}
```

## Steps to reproduce

```powershell
git clone https://github.com/cwe1ss/issue-dotnetcli-restore.git
cd issue-dotnetcli-restore
dotnet restore
dotnet build
```

## Expected  behavior

The build should succeed.

## Actual behavior

`dotnet restore` output. As you can see, the package was found on nuget. I verified that it exists in my `.nuget` folder afterwards.
```
log  : Restoring packages for C:\C3\Repos\cwe1ss\issue-dotnetcli-restore\project.json...
info :   CACHE https://api.nuget.org/v3-flatcontainer/microsoft.bcl/index.json
info :   CACHE https://api.nuget.org/v3-flatcontainer/microsoft.bcl/1.1.9/microsoft.bcl.1.1.9.nupkg
log  : Installing Microsoft.Bcl 1.1.9.
info : Committing restore...
log  : Writing lock file to disk. Path: C:\C3\Repos\cwe1ss\issue-dotnetcli-restore\project.lock.json
log  : C:\C3\Repos\cwe1ss\issue-dotnetcli-restore\project.json
log  : Restore completed in 2342ms.

NuGet Config files used:
    C:\Users\c.weiss\AppData\Roaming\NuGet\NuGet.Config

Feeds used:
    https://api.nuget.org/v3/index.json

Installed:
    1 package(s) to C:\C3\Repos\cwe1ss\issue-dotnetcli-restore\project.json
```
    
`dotnet build` output:
```
Project issue-dotnetcli-restore (.NETFramework,Version=v4.6) will be compiled because expected outputs are missing
Compiling issue-dotnetcli-restore for .NETFramework,Version=v4.6
C:\C3\Repos\cwe1ss\issue-dotnetcli-restore\project.json(8,29): error NU1001: The dependency Microsoft.Bcl >= 1.1.9 could not b
e resolved.

Compilation failed.
    0 Warning(s)
    1 Error(s)

Time elapsed 00:00:00.0174324
```

## Environment data
`dotnet --info` output: 1.0.0-rc2-002610

**Comment by cwe1ss** [comment]
Same error on an AppVeyor build so it seems like it is not related to my machine: https://ci.appveyor.com/project/cwe1ss/issue-dotnetcli-restore/build/3

**Comment by cwe1ss** [comment]
This is blocking me rather badly because `Microsoft.Bcl` is used by so many packages (e.g. almost every Azure package) so I can't get rid of it.

Here's the project.lock.json from my sample. As you can see, some of the folders only contain a `_._` file (which has 0 bytes) instead of actual assemblies. I downloaded the package through a browser from nuget.org and had the same contents there so it seems like the package download is ok.

```json
{
  "locked": false,
  "version": 2,
  "targets": {
    ".NETFramework,Version=v4.6": {
      "Microsoft.Bcl/1.1.9": {
        "type": "package",
        "dependencies": {
          "Microsoft.Bcl.Build": "1.0.14"
        },
        "compile": {
          "lib/net45/_._": {}
        },
        "runtime": {
          "lib/net45/_._": {}
        }
      },
      "Microsoft.Bcl.Build/1.0.14": {
        "type": "package"
      }
    }
  },
  "libraries": {
    "Microsoft.Bcl/1.1.9": {
      "sha512": "USQ55innJy8K+tAXvVa1O8dUTp2s7pmJ5cJj6Tl02HtGc2xBDj0P2QH5620HbGXIWKWYhydoeQF8Rm/JRDBGhw==",
      "type": "package",
      "files": [
        "License-Stable.rtf",
        "Microsoft.Bcl.1.1.9.nupkg.sha512",
        "Microsoft.Bcl.nuspec",
        "content/monoandroid/_._",
        "content/monotouch/_._",
        "content/net45/_._",
        "content/portable-net45+win8+wp8+wpa81/_._",
        "content/portable-net45+win8+wpa81/_._",
        "content/portable-net451+win81+wpa81/_._",
        "content/portable-net451+win81/_._",
        "content/portable-win81+wp81+wpa81/_._",
        "content/sl4/_._",
        "content/sl5/_._",
        "content/win8/_._",
        "content/wp8/_._",
        "content/wpa81/_._",
        "lib/monoandroid/_._",
        "lib/monotouch/_._",
        "lib/net40/System.IO.dll",
        "lib/net40/System.IO.xml",
        "lib/net40/System.Runtime.dll",
        "lib/net40/System.Runtime.xml",
        "lib/net40/System.Threading.Tasks.dll",
        "lib/net40/System.Threading.Tasks.xml",
        "lib/net40/ensureRedirect.xml",
        "lib/net45/_._",
        "lib/portable-net40+sl4+win8+wp71+wpa81/System.IO.dll",
        "lib/portable-net40+sl4+win8+wp71+wpa81/System.IO.xml",
        "lib/portable-net40+sl4+win8+wp71+wpa81/System.Runtime.dll",
        "lib/portable-net40+sl4+win8+wp71+wpa81/System.Runtime.xml",
        "lib/portable-net40+sl4+win8+wp71+wpa81/System.Threading.Tasks.dll",
        "lib/portable-net40+sl4+win8+wp71+wpa81/System.Threading.Tasks.xml",
        "lib/portable-net40+sl4+win8+wp71+wpa81/ensureRedirect.xml",
        "lib/portable-net40+sl4+win8+wp8+wpa81/System.IO.dll",
        "lib/portable-net40+sl4+win8+wp8+wpa81/System.IO.xml",
        "lib/portable-net40+sl4+win8+wp8+wpa81/System.Runtime.dll",
        "lib/portable-net40+sl4+win8+wp8+wpa81/System.Runtime.xml",
        "lib/portable-net40+sl4+win8+wp8+wpa81/System.Threading.Tasks.dll",
        "lib/portable-net40+sl4+win8+wp8+wpa81/System.Threading.Tasks.xml",
        "lib/portable-net40+sl4+win8+wp8+wpa81/ensureRedirect.xml",
        "lib/portable-net40+sl4+win8/System.IO.dll",
        "lib/portable-net40+sl4+win8/System.IO.xml",
        "lib/portable-net40+sl4+win8/System.Runtime.dll",
        "lib/portable-net40+sl4+win8/System.Runtime.xml",
        "lib/portable-net40+sl4+win8/System.Threading.Tasks.dll",
        "lib/portable-net40+sl4+win8/System.Threading.Tasks.xml",
        "lib/portable-net40+sl4+win8/ensureRedirect.xml",
        "lib/portable-net40+sl5+win8+wp8+wpa81/System.IO.dll",
        "lib/portable-net40+sl5+win8+wp8+wpa81/System.IO.xml",
        "lib/portable-net40+sl5+win8+wp8+wpa81/System.Runtime.dll",
        "lib/portable-net40+sl5+win8+wp8+wpa81/System.Runtime.xml",
        "lib/portable-net40+sl5+win8+wp8+wpa81/System.Threading.Tasks.dll",
        "lib/portable-net40+sl5+win8+wp8+wpa81/System.Threading.Tasks.xml",
        "lib/portable-net40+sl5+win8+wp8+wpa81/ensureRedirect.xml",
        "lib/portable-net40+win8+wp8+wpa81/System.IO.dll",
        "lib/portable-net40+win8+wp8+wpa81/System.IO.xml",
        "lib/portable-net40+win8+wp8+wpa81/System.Runtime.dll",
        "lib/portable-net40+win8+wp8+wpa81/System.Runtime.xml",
        "lib/portable-net40+win8+wp8+wpa81/System.Threading.Tasks.dll",
        "lib/portable-net40+win8+wp8+wpa81/System.Threading.Tasks.xml",
        "lib/portable-net40+win8+wp8+wpa81/ensureRedirect.xml",
        "lib/portable-net40+win8/System.IO.dll",
        "lib/portable-net40+win8/System.IO.xml",
        "lib/portable-net40+win8/System.Runtime.dll",
        "lib/portable-net40+win8/System.Runtime.xml",
        "lib/portable-net40+win8/System.Threading.Tasks.dll",
        "lib/portable-net40+win8/System.Threading.Tasks.xml",
        "lib/portable-net40+win8/ensureRedirect.xml",
        "lib/portable-net45+win8+wp8+wpa81/_._",
        "lib/portable-net45+win8+wpa81/_._",
        "lib/portable-net451+win81+wpa81/_._",
        "lib/portable-net451+win81/_._",
        "lib/portable-win81+wp81+wpa81/_._",
        "lib/sl4-windowsphone71/System.IO.dll",
        "lib/sl4-windowsphone71/System.IO.xml",
        "lib/sl4-windowsphone71/System.Runtime.dll",
        "lib/sl4-windowsphone71/System.Runtime.xml",
        "lib/sl4-windowsphone71/System.Threading.Tasks.dll",
        "lib/sl4-windowsphone71/System.Threading.Tasks.xml",
        "lib/sl4-windowsphone71/ensureRedirect.xml",
        "lib/sl4/System.IO.dll",
        "lib/sl4/System.IO.xml",
        "lib/sl4/System.Runtime.dll",
        "lib/sl4/System.Runtime.xml",
        "lib/sl4/System.Threading.Tasks.dll",
        "lib/sl4/System.Threading.Tasks.xml",
        "lib/sl5/System.IO.dll",
        "lib/sl5/System.IO.xml",
        "lib/sl5/System.Runtime.dll",
        "lib/sl5/System.Runtime.xml",
        "lib/sl5/System.Threading.Tasks.dll",
        "lib/sl5/System.Threading.Tasks.xml",
        "lib/win8/_._",
        "lib/wp8/_._",
        "lib/wpa81/_._"
      ]
    },
    "Microsoft.Bcl.Build/1.0.14": {
      "sha512": "cDLKSvNvRa519hplsbSoYqO69TjdDIhfjtKUM0g20/nVROoWsGav9KCI9HtnGjLmdV1+TcUUDhbotcllibjPEA==",
      "type": "package",
      "files": [
        "License-Stable.rtf",
        "Microsoft.Bcl.Build.1.0.14.nupkg.sha512",
        "Microsoft.Bcl.Build.nuspec",
        "content/net40/_._",
        "content/netcore45/_._",
        "content/portable-net40+win8+sl4+wp71+wpa81/_._",
        "content/sl4-windowsphone71/_._",
        "content/sl4/_._",
        "tools/Install.ps1",
        "tools/Microsoft.Bcl.Build.Tasks.dll",
        "tools/Microsoft.Bcl.Build.targets",
        "tools/Uninstall.ps1"
      ]
    }
  },
  "projectFileDependencyGroups": {
    "": [
      "Microsoft.Bcl >= 1.1.9"
    ],
    ".NETFramework,Version=v4.6": []
  },
  "tools": {},
  "projectFileToolGroups": {}
}
```

I'm not familiar with the nuget package schema but I'm pretty sure `dotnet restore` should not take these `_._` files in the `targets/.../compile` section because they are pretty tough to compile :-) 

For this reason I deleted the `_._` file from the "net45" folder and copied the files from "net40" to "net45". After this, `dotnet resolve` resulted in this project.lock.json and `dotnet build` succeeded!

```json
{
  "locked": false,
  "version": 2,
  "targets": {
    ".NETFramework,Version=v4.6": {
      "Microsoft.Bcl/1.1.9": {
        "type": "package",
        "dependencies": {
          "Microsoft.Bcl.Build": "1.0.14"
        },
        "compile": {
          "lib/net45/System.IO.dll": {},
          "lib/net45/System.Runtime.dll": {},
          "lib/net45/System.Threading.Tasks.dll": {}
        },
        "runtime": {
          "lib/net45/System.IO.dll": {},
          "lib/net45/System.Runtime.dll": {},
          "lib/net45/System.Threading.Tasks.dll": {}
        }
      },
      "Microsoft.Bcl.Build/1.0.14": {
        "type": "package"
      }
    }
  },
  "libraries": {
    "Microsoft.Bcl/1.1.9": {
      "sha512": "USQ55innJy8K+tAXvVa1O8dUTp2s7pmJ5cJj6Tl02HtGc2xBDj0P2QH5620HbGXIWKWYhydoeQF8Rm/JRDBGhw==",
      "type": "package",
      "files": [
        "License-Stable.rtf",
        "Microsoft.Bcl.1.1.9.nupkg.sha512",
        "Microsoft.Bcl.nuspec",
        "content/monoandroid/_._",
        "content/monotouch/_._",
        "content/net45/_._",
        "content/portable-net45+win8+wp8+wpa81/_._",
        "content/portable-net45+win8+wpa81/_._",
        "content/portable-net451+win81+wpa81/_._",
        "content/portable-net451+win81/_._",
        "content/portable-win81+wp81+wpa81/_._",
        "content/sl4/_._",
        "content/sl5/_._",
        "content/win8/_._",
        "content/wp8/_._",
        "content/wpa81/_._",
        "lib/monoandroid/_._",
        "lib/monotouch/_._",
        "lib/net40/System.IO.dll",
        "lib/net40/System.IO.xml",
        "lib/net40/System.Runtime.dll",
        "lib/net40/System.Runtime.xml",
        "lib/net40/System.Threading.Tasks.dll",
        "lib/net40/System.Threading.Tasks.xml",
        "lib/net40/ensureRedirect.xml",
        "lib/net45/System.IO.dll",
        "lib/net45/System.IO.xml",
        "lib/net45/System.Runtime.dll",
        "lib/net45/System.Runtime.xml",
        "lib/net45/System.Threading.Tasks.dll",
        "lib/net45/System.Threading.Tasks.xml",
        "lib/net45/ensureRedirect.xml",
        "lib/portable-net40+sl4+win8+wp71+wpa81/System.IO.dll",
        "lib/portable-net40+sl4+win8+wp71+wpa81/System.IO.xml",
        "lib/portable-net40+sl4+win8+wp71+wpa81/System.Runtime.dll",
        "lib/portable-net40+sl4+win8+wp71+wpa81/System.Runtime.xml",
        "lib/portable-net40+sl4+win8+wp71+wpa81/System.Threading.Tasks.dll",
        "lib/portable-net40+sl4+win8+wp71+wpa81/System.Threading.Tasks.xml",
        "lib/portable-net40+sl4+win8+wp71+wpa81/ensureRedirect.xml",
        "lib/portable-net40+sl4+win8+wp8+wpa81/System.IO.dll",
        "lib/portable-net40+sl4+win8+wp8+wpa81/System.IO.xml",
        "lib/portable-net40+sl4+win8+wp8+wpa81/System.Runtime.dll",
        "lib/portable-net40+sl4+win8+wp8+wpa81/System.Runtime.xml",
        "lib/portable-net40+sl4+win8+wp8+wpa81/System.Threading.Tasks.dll",
        "lib/portable-net40+sl4+win8+wp8+wpa81/System.Threading.Tasks.xml",
        "lib/portable-net40+sl4+win8+wp8+wpa81/ensureRedirect.xml",
        "lib/portable-net40+sl4+win8/System.IO.dll",
        "lib/portable-net40+sl4+win8/System.IO.xml",
        "lib/portable-net40+sl4+win8/System.Runtime.dll",
        "lib/portable-net40+sl4+win8/System.Runtime.xml",
        "lib/portable-net40+sl4+win8/System.Threading.Tasks.dll",
        "lib/portable-net40+sl4+win8/System.Threading.Tasks.xml",
        "lib/portable-net40+sl4+win8/ensureRedirect.xml",
        "lib/portable-net40+sl5+win8+wp8+wpa81/System.IO.dll",
        "lib/portable-net40+sl5+win8+wp8+wpa81/System.IO.xml",
        "lib/portable-net40+sl5+win8+wp8+wpa81/System.Runtime.dll",
        "lib/portable-net40+sl5+win8+wp8+wpa81/System.Runtime.xml",
        "lib/portable-net40+sl5+win8+wp8+wpa81/System.Threading.Tasks.dll",
        "lib/portable-net40+sl5+win8+wp8+wpa81/System.Threading.Tasks.xml",
        "lib/portable-net40+sl5+win8+wp8+wpa81/ensureRedirect.xml",
        "lib/portable-net40+win8+wp8+wpa81/System.IO.dll",
        "lib/portable-net40+win8+wp8+wpa81/System.IO.xml",
        "lib/portable-net40+win8+wp8+wpa81/System.Runtime.dll",
        "lib/portable-net40+win8+wp8+wpa81/System.Runtime.xml",
        "lib/portable-net40+win8+wp8+wpa81/System.Threading.Tasks.dll",
        "lib/portable-net40+win8+wp8+wpa81/System.Threading.Tasks.xml",
        "lib/portable-net40+win8+wp8+wpa81/ensureRedirect.xml",
        "lib/portable-net40+win8/System.IO.dll",
        "lib/portable-net40+win8/System.IO.xml",
        "lib/portable-net40+win8/System.Runtime.dll",
        "lib/portable-net40+win8/System.Runtime.xml",
        "lib/portable-net40+win8/System.Threading.Tasks.dll",
        "lib/portable-net40+win8/System.Threading.Tasks.xml",
        "lib/portable-net40+win8/ensureRedirect.xml",
        "lib/portable-net45+win8+wp8+wpa81/_._",
        "lib/portable-net45+win8+wpa81/_._",
        "lib/portable-net451+win81+wpa81/_._",
        "lib/portable-net451+win81/_._",
        "lib/portable-win81+wp81+wpa81/_._",
        "lib/sl4-windowsphone71/System.IO.dll",
        "lib/sl4-windowsphone71/System.IO.xml",
        "lib/sl4-windowsphone71/System.Runtime.dll",
        "lib/sl4-windowsphone71/System.Runtime.xml",
        "lib/sl4-windowsphone71/System.Threading.Tasks.dll",
        "lib/sl4-windowsphone71/System.Threading.Tasks.xml",
        "lib/sl4-windowsphone71/ensureRedirect.xml",
        "lib/sl4/System.IO.dll",
        "lib/sl4/System.IO.xml",
        "lib/sl4/System.Runtime.dll",
        "lib/sl4/System.Runtime.xml",
        "lib/sl4/System.Threading.Tasks.dll",
        "lib/sl4/System.Threading.Tasks.xml",
        "lib/sl5/System.IO.dll",
        "lib/sl5/System.IO.xml",
        "lib/sl5/System.Runtime.dll",
        "lib/sl5/System.Runtime.xml",
        "lib/sl5/System.Threading.Tasks.dll",
        "lib/sl5/System.Threading.Tasks.xml",
        "lib/win8/_._",
        "lib/wp8/_._",
        "lib/wpa81/_._"
      ]
    },
    "Microsoft.Bcl.Build/1.0.14": {
      "sha512": "cDLKSvNvRa519hplsbSoYqO69TjdDIhfjtKUM0g20/nVROoWsGav9KCI9HtnGjLmdV1+TcUUDhbotcllibjPEA==",
      "type": "package",
      "files": [
        "License-Stable.rtf",
        "Microsoft.Bcl.Build.1.0.14.nupkg.sha512",
        "Microsoft.Bcl.Build.nuspec",
        "content/net40/_._",
        "content/netcore45/_._",
        "content/portable-net40+win8+sl4+wp71+wpa81/_._",
        "content/sl4-windowsphone71/_._",
        "content/sl4/_._",
        "tools/Install.ps1",
        "tools/Microsoft.Bcl.Build.Tasks.dll",
        "tools/Microsoft.Bcl.Build.targets",
        "tools/Uninstall.ps1"
      ]
    }
  },
  "projectFileDependencyGroups": {
    "": [
      "Microsoft.Bcl >= 1.1.9"
    ],
    ".NETFramework,Version=v4.6": []
  },
  "tools": {},
  "projectFileToolGroups": {}
}
```

So it seems like `dotnet restore` does not resolve it correctly.

**Comment by cwe1ss** [comment]
Same happens for other libraries so this is not special to `Microsoft.Bcl`. Here's an example snippet from a project.lock.json that depends on a recent System.* library:

```json
"System.Collections/4.0.11-rc2-24027": {
  "type": "package",
  "compile": {
    "ref/net45/_._": {}
  },
  "runtime": {
    "lib/net45/_._": {}
  }
}
```

**Comment by piotrpMSFT** [comment]
@cwe1ss thanks for the detailed bug report! The restore operation + lock file creation are handled by NuGet. Can you open this bug in https://github.com/nuget/home?

/cc @emgarten

**Comment by cwe1ss** [comment]
re-opened in https://github.com/NuGet/Home/issues/2715

**Comment by cwe1ss** [comment]
Re-opened after discussion with @emgarten - this seems to be an issue with `dotnet build`. I will re-post my comments from there for clarity.

**Comment by cwe1ss** [comment]
When I delete the `_._` file from `.nuget\packages\Microsoft.Bcl\1.1.9\lib\net45` and do a `dotnet restore`, the project.lock.json takes the files from net40 and the build succeeds! 

I don't know the meaning of these `_._` files so I'm out of further ideas. Some tips for a quick fix would be great!

```json
{
  "locked": false,
  "version": 2,
  "targets": {
    ".NETFramework,Version=v4.6": {
      "Microsoft.Bcl/1.1.9": {
        "type": "package",
        "dependencies": {
          "Microsoft.Bcl.Build": "1.0.14"
        },
        "compile": {
          "lib/net40/System.IO.dll": {},
          "lib/net40/System.Runtime.dll": {},
          "lib/net40/System.Threading.Tasks.dll": {}
        },
        "runtime": {
          "lib/net40/System.IO.dll": {},
          "lib/net40/System.Runtime.dll": {},
          "lib/net40/System.Threading.Tasks.dll": {}
        }
      },
      "Microsoft.Bcl.Build/1.0.14": {
        "type": "package"
      }
    }
  },
  "libraries": {
    "Microsoft.Bcl/1.1.9": {
      "sha512": "USQ55innJy8K+tAXvVa1O8dUTp2s7pmJ5cJj6Tl02HtGc2xBDj0P2QH5620HbGXIWKWYhydoeQF8Rm/JRDBGhw==",
      "type": "package",
      "files": [
        "License-Stable.rtf",
        "Microsoft.Bcl.1.1.9.nupkg.sha512",
        "Microsoft.Bcl.nuspec",
        "content/monoandroid/_._",
        "content/monotouch/_._",
        "content/net45/_._",
        "content/portable-net45+win8+wp8+wpa81/_._",
        "content/portable-net45+win8+wpa81/_._",
        "content/portable-net451+win81+wpa81/_._",
        "content/portable-net451+win81/_._",
        "content/portable-win81+wp81+wpa81/_._",
        "content/sl4/_._",
        "content/sl5/_._",
        "content/win8/_._",
        "content/wp8/_._",
        "content/wpa81/_._",
        "lib/monoandroid/_._",
        "lib/monotouch/_._",
        "lib/net40/System.IO.dll",
        "lib/net40/System.IO.xml",
        "lib/net40/System.Runtime.dll",
        "lib/net40/System.Runtime.xml",
        "lib/net40/System.Threading.Tasks.dll",
        "lib/net40/System.Threading.Tasks.xml",
        "lib/net40/ensureRedirect.xml",
        "lib/portable-net40+sl4+win8+wp71+wpa81/System.IO.dll",
        "lib/portable-net40+sl4+win8+wp71+wpa81/System.IO.xml",
        "lib/portable-net40+sl4+win8+wp71+wpa81/System.Runtime.dll",
        "lib/portable-net40+sl4+win8+wp71+wpa81/System.Runtime.xml",
        "lib/portable-net40+sl4+win8+wp71+wpa81/System.Threading.Tasks.dll",
        "lib/portable-net40+sl4+win8+wp71+wpa81/System.Threading.Tasks.xml",
        "lib/portable-net40+sl4+win8+wp71+wpa81/ensureRedirect.xml",
        "lib/portable-net40+sl4+win8+wp8+wpa81/System.IO.dll",
        "lib/portable-net40+sl4+win8+wp8+wpa81/System.IO.xml",
        "lib/portable-net40+sl4+win8+wp8+wpa81/System.Runtime.dll",
        "lib/portable-net40+sl4+win8+wp8+wpa81/System.Runtime.xml",
        "lib/portable-net40+sl4+win8+wp8+wpa81/System.Threading.Tasks.dll",
        "lib/portable-net40+sl4+win8+wp8+wpa81/System.Threading.Tasks.xml",
        "lib/portable-net40+sl4+win8+wp8+wpa81/ensureRedirect.xml",
        "lib/portable-net40+sl4+win8/System.IO.dll",
        "lib/portable-net40+sl4+win8/System.IO.xml",
        "lib/portable-net40+sl4+win8/System.Runtime.dll",
        "lib/portable-net40+sl4+win8/System.Runtime.xml",
        "lib/portable-net40+sl4+win8/System.Threading.Tasks.dll",
        "lib/portable-net40+sl4+win8/System.Threading.Tasks.xml",
        "lib/portable-net40+sl4+win8/ensureRedirect.xml",
        "lib/portable-net40+sl5+win8+wp8+wpa81/System.IO.dll",
        "lib/portable-net40+sl5+win8+wp8+wpa81/System.IO.xml",
        "lib/portable-net40+sl5+win8+wp8+wpa81/System.Runtime.dll",
        "lib/portable-net40+sl5+win8+wp8+wpa81/System.Runtime.xml",
        "lib/portable-net40+sl5+win8+wp8+wpa81/System.Threading.Tasks.dll",
        "lib/portable-net40+sl5+win8+wp8+wpa81/System.Threading.Tasks.xml",
        "lib/portable-net40+sl5+win8+wp8+wpa81/ensureRedirect.xml",
        "lib/portable-net40+win8+wp8+wpa81/System.IO.dll",
        "lib/portable-net40+win8+wp8+wpa81/System.IO.xml",
        "lib/portable-net40+win8+wp8+wpa81/System.Runtime.dll",
        "lib/portable-net40+win8+wp8+wpa81/System.Runtime.xml",
        "lib/portable-net40+win8+wp8+wpa81/System.Threading.Tasks.dll",
        "lib/portable-net40+win8+wp8+wpa81/System.Threading.Tasks.xml",
        "lib/portable-net40+win8+wp8+wpa81/ensureRedirect.xml",
        "lib/portable-net40+win8/System.IO.dll",
        "lib/portable-net40+win8/System.IO.xml",
        "lib/portable-net40+win8/System.Runtime.dll",
        "lib/portable-net40+win8/System.Runtime.xml",
        "lib/portable-net40+win8/System.Threading.Tasks.dll",
        "lib/portable-net40+win8/System.Threading.Tasks.xml",
        "lib/portable-net40+win8/ensureRedirect.xml",
        "lib/portable-net45+win8+wp8+wpa81/_._",
        "lib/portable-net45+win8+wpa81/_._",
        "lib/portable-net451+win81+wpa81/_._",
        "lib/portable-net451+win81/_._",
        "lib/portable-win81+wp81+wpa81/_._",
        "lib/sl4-windowsphone71/System.IO.dll",
        "lib/sl4-windowsphone71/System.IO.xml",
        "lib/sl4-windowsphone71/System.Runtime.dll",
        "lib/sl4-windowsphone71/System.Runtime.xml",
        "lib/sl4-windowsphone71/System.Threading.Tasks.dll",
        "lib/sl4-windowsphone71/System.Threading.Tasks.xml",
        "lib/sl4-windowsphone71/ensureRedirect.xml",
        "lib/sl4/System.IO.dll",
        "lib/sl4/System.IO.xml",
        "lib/sl4/System.Runtime.dll",
        "lib/sl4/System.Runtime.xml",
        "lib/sl4/System.Threading.Tasks.dll",
        "lib/sl4/System.Threading.Tasks.xml",
        "lib/sl5/System.IO.dll",
        "lib/sl5/System.IO.xml",
        "lib/sl5/System.Runtime.dll",
        "lib/sl5/System.Runtime.xml",
        "lib/sl5/System.Threading.Tasks.dll",
        "lib/sl5/System.Threading.Tasks.xml",
        "lib/win8/_._",
        "lib/wp8/_._",
        "lib/wpa81/_._"
      ]
    },
    "Microsoft.Bcl.Build/1.0.14": {
      "sha512": "cDLKSvNvRa519hplsbSoYqO69TjdDIhfjtKUM0g20/nVROoWsGav9KCI9HtnGjLmdV1+TcUUDhbotcllibjPEA==",
      "type": "package",
      "files": [
        "License-Stable.rtf",
        "Microsoft.Bcl.Build.1.0.14.nupkg.sha512",
        "Microsoft.Bcl.Build.nuspec",
        "content/net40/_._",
        "content/netcore45/_._",
        "content/portable-net40+win8+sl4+wp71+wpa81/_._",
        "content/sl4-windowsphone71/_._",
        "content/sl4/_._",
        "tools/Install.ps1",
        "tools/Microsoft.Bcl.Build.Tasks.dll",
        "tools/Microsoft.Bcl.Build.targets",
        "tools/Uninstall.ps1"
      ]
    }
  },
  "projectFileDependencyGroups": {
    "": [
      "Microsoft.Bcl >= 1.1.9"
    ],
    ".NETFramework,Version=v4.6": []
  },
  "tools": {},
  "projectFileToolGroups": {}
}
```

**Comment by cwe1ss** [comment]
@piotrpMSFT Are you sure this isn't an issue of `dotnet build`/`dotnet cli`? I tried to find out what `_._` means and found thid quote by @emgarten from https://github.com/dotnet/cli/issues/2250#issuecomment-206704507

> The placeholder `_._` is added to the lock file so that compatibility can be checked outside of NuGet. Looks like this has already been fixed, but it is expected that consumers of the lock file understand the `_._` convention.

This sounds like `dotnet build` should be able to deal with this. We haven't had issues with this until recently so maybe a recent change somewhere in dotnet cli broke this?!?

This issue is really bad for us because we can't compile anymore - we had to fallback to dnx :(

-----------------------

@piotrpMSFT Are you sure this isn't an issue of `dotnet build`/`dotnet cli`? I tried to find out what `_._` means and found thid quote by @emgarten from https://github.com/dotnet/cli/issues/2250#issuecomment-206704507

> The placeholder `_._` is added to the lock file so that compatibility can be checked outside of NuGet. Looks like this has already been fixed, but it is expected that consumers of the lock file understand the `_._` convention.

This sounds like `dotnet build` should be able to deal with this. We haven't had issues with this until recently so maybe a recent change somewhere in dotnet cli broke this?!?

This issue is really bad for us because we can't compile anymore - we had to fallback to dnx :(

**Comment by cwe1ss** [comment]
I tried this with a few older CLI versions:

* 1.0.0-rc2-002439: success
* 1.0.0-rc2-002457: success
* 1.0.0-rc2-002474: failure

So it seems like this is an issue in dotnet/cli. Unfortunately, I don't have any version in between and I also don't know to which commit messages they relate so I'm having a hard time narrowing it further down.

**Comment by piotrpMSFT** [comment]
I chatted about this with @ericstj offline and it seems this library should no longer be used. @ericstj can you add some detail here?

**Comment by cwe1ss** [comment]
Hi Piotr, I understand that and I would not use it for my own libraries. Unfortunately, this library is quite popular on NuGet. One example is [Microsoft.Azure.Common](https://www.nuget.org/packages/Microsoft.Azure.Common/) which is used e.g. by the [Azure Key Vault Library](https://www.nuget.org/packages/Microsoft.Azure.KeyVault/).

They are not open-source so I can't even create a PR/fork. If those libraries don't get updated we are *completely* blocked. I have no idea what I should do in that case because we can't get rid of Azure Key Vault.

I guess many people will be affected by this so I really hope you add a fix for this.

**Comment by ericstj** [comment]
In this case it is OK to use it.  The project is targeting desktop.  Before I was mentioning how this package shouldn't be used when targeting NETCoreApp.

The `_._` are a nuget mechanism to indicate that the package is supported on that platform and doesn't need to provide any assets.  In the case of BCL this is because the whole purpose of that package is to add DLLs to other platforms (Silverlight, Windows Phone) that were already part of `net45`.

I can reproduce the problem, the interesting thing is that it doesn't repro for restore, only for build.  I suspect this is another problem caused by https://github.com/dotnet/cli/pull/2597.

/cc @davidfowl

**Comment by cwe1ss** [comment]
This is not an accurate research, but Google yields more than 4700 results for this search query: ["Microsoft.Bcl" site:nuget.org/packages](https://www.google.at/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=%22Microsoft.Bcl%22+site:nuget.org%2Fpackages). Even if there are a lot less packages that depend on that, it's still a lot.

**Comment by ericstj** [comment]
The logic here is busted:
https://github.com/dotnet/cli/pull/2597/files#diff-5c8786bd7f881c627f83a5a9b6620c07R388

Just because a package has a placeholder doesn't mean it represents an inbox facade.  Placeholders have been around in NuGet for a long time and folks use them for more reasons than representing an inbox asset.  I think it the framework reference is unresolved then CLI shouldn't muck with replacing the package with an unresolved framework reference.  It should assume that the package is not using a placeholder to represent inbox facade and let it stay resolved without contributing anything to the build.

**Comment by piotrpMSFT** [comment]
@davidfowl can you either provide a fix or guidance for a fix here? I know that this topic area is important to you...

**Comment by cwe1ss** [comment]
This is fixed in the current version (1.0.0-preview1-002700)! Is there anything left you want to do on this issue? Feel free to close it, if not.

thanks guys!!
